libwarpaffine/sliceswriter/SlicesWriterTbb.cpp

```diff
@@ -30,6 +33,18 @@ CziSlicesWriterTbb::CziSlicesWriterTbb(AppContext& context, const std::wstring&
 
 void CziSlicesWriterTbb::AddSlice(const AddSliceInfo& add_slice_info)
 {
+    if (this->use_acquisition_tiles_) 
+    {
+        int z;
+        add_slice_info.coordinate.TryGetPosition(libCZI::DimensionIndex::Z, &z);
+        std::lock_guard<std::mutex> lock(this->retiling_mutex_); // unlocks automatically at end of scope
```

Comment:
not sure, but I wonder - we are locking here for exclusive access to retilingIds_, and we may mutate the map (line 44), but we don't seem to lock when accessing the map in line 98. Granted, we only have a read-access in line 98, but I don't see this read-access synchronized with the write-access in line 44, so they could execute concurrently and result in crash/undefined behavior. Or, am I missing something?
Best case we'd want to have a reader-writer-lock here, or we could TBB's [concurrent_hash_map ](https://www.intel.com/content/www/us/en/docs/onetbb/developer-guide-api-reference/2021-6/concurrent-hash-map.html) a spin here.

Reply:
you are probably right. I was trusting my two debugging sessions where all the writing was done way before any of the reading, but I'll make it safe anyway.

Reply:
the whole structure became obsolete now.